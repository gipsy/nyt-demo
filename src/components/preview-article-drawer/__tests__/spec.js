import React from 'react'
import ReactDOM from 'react-dom'
import { render, fireEvent } from '@tests/utils/render-with-app-context'
import { initialData } from '@tests/initial-data'

import PreviewArticleDrawer from '@components/preview-article-drawer'

describe('<PreviewArticleDrawer />', () => {
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element, node) => {
      return element
    })
  })

  afterEach(() => {
    ReactDOM.createPortal.mockClear()
  })

  test('should match snapshot', () => {
    const context = {...initialData, news: { ...initialData.news, drawer: true }}
    const { container } = render(<PreviewArticleDrawer />, { context })
    expect(container).toMatchSnapshot()
  })

  test('should not have "ant-drawer-open" class on close button clicked', () => {
    const context = {...initialData, news: { ...initialData.news, drawer: true }}
    const { container } = render(<PreviewArticleDrawer />, { context })

    const root = container.querySelector('.ant-drawer')
    const closeButton = container.querySelector('[aria-label="Close"]')
    fireEvent.click(closeButton)
    expect(root).not.toHaveClass('ant-drawer-open')
  })
})
