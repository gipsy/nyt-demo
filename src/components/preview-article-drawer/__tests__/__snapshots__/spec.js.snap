// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreviewArticleDrawer /> should match snapshot 1`] = `
.c0 {
  font-family: 'Noto Serif KR',sans-serif;
  font-size: 28px;
  color: #000;
}

.c3 {
  display: inline-block;
  background-color: #ececec;
  width: 100%;
}

.c4 {
  font-size: 16px;
  color: #000;
  margin-top: 20px;
}

.c1 {
  text-transform: uppercase;
  font-size: 12px;
  color: #ddd;
  display: inline-block;
}

.c2 {
  font-family: 'Helvetica',sans-serif;
  font-weight: 500;
  font-size: 14px;
  display: inline-block;
  float: right;
}

@media (max-width:600px) {
  .c0 {
    font-size: 23px;
  }
}

<div>
  <div
    class="ant-drawer ant-drawer-right ant-drawer-open"
  >
    <div
      class="ant-drawer-mask"
    />
    <div
      class="ant-drawer-content-wrapper"
      style="width: 50%;"
    >
      <div
        class="ant-drawer-content"
      >
        <div
          class="ant-drawer-wrapper-body"
          style="overflow: auto; height: 100%;"
        >
          <button
            aria-label="Close"
            class="ant-drawer-close"
          >
            <span
              class="ant-drawer-close-x"
            >
              <i
                class="anticon anticon-close"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="close"
                  fill="currentColor"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                  />
                </svg>
              </i>
            </span>
          </button>
          <div
            class="ant-drawer-body"
          >
            <h2
              class="c0"
            >
              Singapore, for Those Who Aren’t Crazy Rich
            </h2>
            <div
              class="ant-divider ant-divider-horizontal"
            />
            <span
              class="c1"
            >
              The New York Times
            </span>
            <time
              class="c2"
            >
              Sep 21, 2018
            </time>
            <img
              class="c3"
              src="https://graphics8.nytimes.com/images/2018/09/23/travel/23singapore1/merlin_142944708_9276e9dd-b36e-40f3-90f6-34e79e64552e-jumbo.jpg"
            />
            <div
              class="ant-row"
            >
              <div
                class="ant-col-24"
              >
                <p
                  class="c4"
                >
                  In the hit movie “Crazy Rich Asians,” the country looks luxe. But it’s easy to have a great time (and eat memorably well) without spending much.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
